### DISCUSSION TITLE
[Solved] Cant use Harden-Windows-Security

### DISCUSSION THREAD

**Discussion by 26zl** [HotCakeX/Harden-Windows-Security#306] [Q&A] [answered]
Hi! Im trying to use ur script but it doesnt work, i have used ur script multiple times before on this same laptop but now i cant make it work. Here is what comes up when trying to use the script. 

![451435799_472068409079512_4113104742741025676_n](https://github.com/user-attachments/assets/a754663d-74f3-42d1-8875-95127ab6af5c)
![449828170_875730994369452_2545709538645028250_n](https://github.com/user-attachments/assets/393dace4-306a-48d7-a61c-335ebfc3143f)

**Comment by HotCakeX** [comment]
Hi,
covered that here
https://github.com/HotCakeX/Harden-Windows-Security/discussions/305
fix coming soon

**Reply by 26zl** [reply]
Thank you! Make sense. But i also tried to use this command for a workaround as i understood the problem. 

Set-Culture -CultureInfo 'en-US'

But still didnt work. Take your time with the fix no hassle you have a great script. And has helped me loads. And sure it has helped many others too! Be proud🤩

**Reply by HotCakeX** [reply]
Thank you for the kind words ^^ and sorry i didn't catch this issue sooner 🙏

I fixed it and you can install the new version using this command
```powershell
Install-Module -Name 'Harden-Windows-Security-Module' -Force
```
the reason it took a day for me to release this update is that i had lots of changes and new features planned for version 0.5.0 and i thought there would be enough time to implement them and fix this issue, but eventually i realized i would need more time to perform validation on the new codes so instead of rushing/risking it, i released v0.5.0 as a small update that only fixes this issue. all those new features will come in v0.5.1 after proper validation.

I've learned from this though and will definitely change things so that things like this won't happen again.

**Reply by 26zl** [reply]
Thank you so much for the quick fix! 😊 Your dedication and hard work really show. 

I’ve installed the new version using the command you provided, and everything works perfectly!

I’m looking forward to all the exciting new features in v0.5.1. 

And as always thank you for everything. Keep up the great work!

**Reply by HotCakeX** [reply]
Thank you 😊😊

**Reply by 26zl** [reply]
Hi again! Quick question. How do I unprotect only one category? I have some issues with certain programs that don’t update because they are being blocked. I think it’s because of attack reduction rules.

**Reply by HotCakeX** [reply, marked as answer]
There is an ASR rule that can block programs that were just released and have no history on the threat radar, if your program is blocked because of that you will see a notification allowing you to unblock it with 1 click/tap as Admin.

Other than that, the module doesn't currently offer a way to undo every individual category, there are some categories that can be undone but the rest must be undone altogether.

https://github.com/HotCakeX/Harden-Windows-Security/wiki/Harden%E2%80%90Windows%E2%80%90Security%E2%80%90Module#unprotect-windowssecurity-cmdlet

I'm planning to add this feature in a future update but for now you can go to Group policy editor and turn off any ASR rules you want, or you can turn them all off.

https://learn.microsoft.com/en-us/defender-endpoint/attack-surface-reduction-rules-reference?view=o365-worldwide#asr-rule-to-guid-matrix

<br>

<img width="1280" alt="image" src="https://github.com/user-attachments/assets/b0f3857d-85f8-4fee-b9fd-013782024eeb">

**Reply by HotCakeX** [reply]
The capability to configure ASR rules individually has been added in the latest update
https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/Hardening-Module-v.0.5.9

**Reply by 26zl** [reply]
Amazing work!🤩 Will test the new functions this week, thank you so much👏🏼

May ask. How did you learn powershell? Do you have any good easy to learn resources? I know batch already. 
Thanks in advance!

**Reply by HotCakeX** [reply]
Thanks ^^
I learned it by just using different cmdlets, reading their docs on MS Learn, asking AI, etc.. I've made lots of learning materials here in the wiki, check them out :)
https://github.com/HotCakeX/Harden-Windows-Security/wiki

**Reply by 26zl** [reply]
Thanks a lot😍
